src/library/LoansWithSettlement.sol

```diff
@@ -19,4 +19,69 @@ library LoansWithSettlement {
         /// funds from all loans have been made available.
         bytes settlement;
     }
+
+    /// @notice The number of loans in the input.
+    /// @param loansWithSettlement The list of loans with settlement.
+    /// @return Number of loans in the input.
+    function loanCount(Data memory loansWithSettlement) internal pure returns (uint256) {
+        return loansWithSettlement.loans.length;
+    }
+
+    /// @notice A collision-resistent identifier for the input list of loans
+    /// with settlement.
+    /// @param loansWithSettlement The list of loans with settlement to hash.
+    /// @return A collision-resistent identifier for the input.
+    function hash(Data memory loansWithSettlement) internal pure returns (bytes32) {
+        return keccak256(abi.encode(loansWithSettlement));
+    }
+
+    /// @notice Store the list of loans and the settlement in a format
+    /// expected by this library.
+    /// @param loans List of requested loans.
+    /// @param settlement ABI-encoded settlement call data.
+    /// @return An encoded representation of the input parameters.
+    function encode(Loan.Data[] calldata loans, bytes calldata settlement) internal pure returns (Data memory) {
+        // We invert the order of the input for more efficient popping of the
+        // next array element.
+        Loan.Data[] memory _loans = new Loan.Data[](loans.length);
+        for (uint256 i = 0; i < loans.length; i++) {
+            _loans[i] = loans[loans.length - i - 1];
+        }
+        return Data({settlement: settlement, loans: _loans});
+    }
+
+    /// @notice Remove the next loan that is to be processed from the encoded
+    /// input data and return its parameter.
+    /// @dev The element are popped from the first to the last in the order they
+    /// were presented *before encoding*.
+    /// @param loansWithSettlement The encoded data from which to remove the
+    /// next loan.
+    /// @return amount The amount to be borrowed (see `Loan.Data`).
+    /// @return borrower The address of the borrower contract (see `Loan.Data`).
+    /// @return lender The lender address (see `Loan.Data`).
+    /// @return token The token to borrow (see `Loan.Data`).
+    function popLoan(Data memory loansWithSettlement)
```

Comment:
Should we document our assumption about the loan count > 0? Would we revert in case we try to pop from an empty array (might be good to add a test)